### ISSUE TITLE
systemd should allow to skip "dropping addresses" on restart

### ISSUE THREAD

**Issue by r7vme** [systemd/systemd#10660]
**systemd version the issue has been seen with**
238-coreos and `master` branch (Nov 6)

**Used distribution**

CoreOS 1855.4.0

**Expected behaviour you didn't see**

Statically assigned (by networkd) addresses stay intact on systemd-networkd restart.

**Unexpected behaviour you saw**

sysmemd-networkd removes all "foreign" addresses, even if these are addresses it assigns.

**Steps to reproduce the problem**

1. Create static network configuraion (like below).
2. Enable debug mode (like [here](https://coreos.com/os/docs/latest/network-config-with-networkd.html#debugging-networkd))
3. `systemctl restart systemd-networkd`
4. `journalctl -u systemd-networkd | grep Removing`

Logs will contain.
```
Removing address: 10.0.4.101/24 (valid forever)
```

```
[Match]
Name=eth0
[Network]
Address=10.0.4.101/24
```

**Additional info**

Workaround that worked for me (really ugly) is to add following to `.network` file.

```
[DHCP]
CriticalConnection=yes
```

[Which allows to skip link_drop_foreign_config](https://github.com/systemd/systemd/blob/v238/src/network/networkd-link.c#L2599).

**Comment by poettering** [comment]
Coincidentally fixed 1min ago with #10597. Closing.

**Comment by r7vme** [comment]
LOL :)
